gems/aws-sdk-core/lib/aws-sdk-core/plugins/protocols/rest_xml.rb

```diff
@@ -5,9 +5,21 @@ module Plugins
     module Protocols
       class RestXml < Seahorse::Client::Plugin
 
+        class ContentTypeHandler < Seahorse::Client::Handler
+          def call(context)
+            body = context.http_request.body
+            if !body.respond_to?(:size) ||
+               (body.respond_to?(:size) && body.size > 0)
+              context.http_request.headers['Content-Type'] ||=
+                'application/xml'
+            end
+            @handler.call(context)
+          end
+        end
+
         handler(Rest::Handler)
+        handler(ContentTypeHandler, priority: 30)
```

Comment:
I'd move this priority closer to the builder. You may be able to just remove the priority as the default might be good enough - if it's out of order, you can register it before the rest handler.